feat: implement discussion mute/unmute feature with user and staff-le…

naincy128 opened the pull request:
This pull request adds a user muting feature in discussion forums, including personal and course-wide mutes, mute exceptions, and moderation audit logs. Three new models (DiscussionUserMute, DiscussionMuteException, DiscussionModerationLog) are added with admin interfaces, permissions, and indexing. Muted content is automatically filtered across threads, comments, search results, and profiles.

naincy128 committed:
feat: implement discussion mute/unmute feature with user and staff-level controls

openedx-webhooks commented:
Thanks for the pull request, @naincy128!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>


<details><summary>:radio_button: Submit a signed contributor agreement (CLA)</summary>

:warning: We ask all contributors to the Open edX project to submit a [signed contributor agreement](https://openedx.org/cla) or indicate their institutional affiliation.
Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md) file for more information.

If you've signed an agreement in the past, you may need to re-sign.
See [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/) for details.

Once you've signed the CLA, please allow 1 business day for it to be processed.
After this time, you can re-run the CLA check by adding a comment below that you have signed it.
If the CLA check continues to fail, you can tag the `@openedx/cla-problems` team in a comment for further assistance.
</details>
<!-- comment:no_cla -->




<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary>:radio_button: Update the status of your PR</summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details>
<!-- comment:end_of_wip -->


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

naincy128 committed:
Fix migration issue: Remove duplicate timestamp fields migration

- Removed 0011_add_timestamped_fields_to_moderationlog.py (DiscussionModerationLog already inherits from TimeStampedModel)
- Removed conflicting merge migration
- Fixed 0010_discussion_muting_models.py to properly create all mute-related tables
- This resolves the 'Duplicate column name created' error

naincy128 committed:
feat: implement discussion mute/unmute feature with user and staff-level controls
